hackathon_site/hardware/test_api.py

```diff
@@ -188,9 +188,606 @@ def test_hardware_get_success(self):
+            notes="notes",
+            quantity_available=10,
+            max_per_team=1,
+            picture="/picture/location",
+        )
+        limited_hardware.categories.add(self.category_limit_10.pk)
+
+        limited_category_hardware = Hardware.objects.create(
+            name="name",
+            model_number="model",
+            manufacturer="manufacturer",
+            datasheet="/datasheet/location/",
+            notes="notes",
+            quantity_available=10,
+            max_per_team=10,
+            picture="/picture/location",
+        )
+        limited_category_hardware.categories.add(self.category_limit_1.pk)
+
+        request_data = {
+            "hardware": [
+                {"id": limited_hardware.id, "quantity": 10},
+                {"id": limited_category_hardware.id, "quantity": 10},
+            ]
+        }
+        response = self.client.post(self.view, request_data, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        pass
+
+    def test_multiple_hardware_success(self):
+        self._login()
+        profile = self._make_event_profile()
+
+        hardware_1 = Hardware.objects.create(
+            name="name",
+            model_number="model",
+            manufacturer="manufacturer",
+            datasheet="/datasheet/location/",
+            notes="notes",
+            quantity_available=10,
+            max_per_team=10,
+            picture="/picture/location",
+        )
+        hardware_1.categories.add(self.category_limit_10.pk)
+
+        hardware_2 = Hardware.objects.create(
+            name="name",
+            model_number="model",
+            manufacturer="manufacturer",
+            datasheet="/datasheet/location/",
+            notes="notes",
+            quantity_available=10,
+            max_per_team=4,
+            picture="/picture/location",
+        )
+        hardware_2.categories.add(self.category_limit_4.pk)
+
+        num_hardware_1_requested = 6
+        num_hardware_2_requested = 4
+
+        request_data = {
+            "hardware": [
+                {"id": hardware_1.id, "quantity": num_hardware_1_requested},
+                {"id": hardware_2.id, "quantity": num_hardware_2_requested},
+            ]
+        }
+        response = self.client.post(self.view, request_data, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        order_id = response.json().get("order_id")
+        self.assertIsNotNone(order_id, "No order id returned")
+
+        order = Order.objects.get(pk=order_id)
+        self.assertCountEqual(order.hardware_set.all(), [hardware_1, hardware_2])
+        self.assertEqual(
+            order.items.filter(hardware=hardware_1).count(), num_hardware_1_requested
+        )
+        self.assertEqual(
+            order.items.filter(hardware=hardware_2).count(), num_hardware_2_requested
+        )
+
+        response_hardware = response.json().get("hardware")
```

Comment:
Same as mentioned above, it would be clearer just to hard code the expected response json and compare the response with that